crates/ruff/src/rules/pycodestyle/rules/doc_line_too_long.rs

```diff
@@ -54,11 +54,24 @@ pub fn doc_line_too_long(lineno: usize, line: &str, settings: &Settings) -> Opti
         settings.pycodestyle.ignore_overlong_task_comments,
         &settings.task_tags,
     ) {
+        let mut start_column = 0;
+        let mut width = 0;
+        let mut length = 0;
+
+        for c in line.chars() {
+            if width < limit {
+                width += c.width().unwrap_or(0);
+                start_column += 1;
+            } else {
+                length += 1;
+            }
+        }
```

Comment:
I wonder if we should move this computation into `is_overlong` somehow, so that it's guaranteed to yield a consistent result with `line_width`?

Reply:
Done. `is_overlong` now does all the calculations and returns an `Option<Overlong>` to give the caller access to the computed range and width.